src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,7 +16,9 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> optionalUser = userService.findByLogin(login);
-        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(password)) {
+        byte[] salt = optionalUser.get().getSalt();
+        String saltedPassword = HashUtil.hashPassword(password, salt);
+        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(saltedPassword)) {
```

Comment:
```suggestion
        if (optionalUser.isPresent() && isPasswordValid(optionalUser.get(), password) {
```
let's create a private method where we will check if the password is valid

Reply:
Added private method isPasswordValid()

Reply:
Now we do check isPresent() before && and after that calling get() on optional